pkg/kapp/cmd/app/deploy.go

```diff
@@ -51,7 +52,12 @@ func NewDeployCmd(o *DeployOptions, flagsFactory cmdcore.FlagsFactory) *cobra.Co
 		Use:     "deploy",
 		Aliases: []string{"d", "dep"},
 		Short:   "Deploy app",
-		RunE:    func(_ *cobra.Command, _ []string) error { return o.Run() },
+		RunE: func(_ *cobra.Command, _ []string) error {
+			if o.ApplyFlags.ServerSideApply {
+				o.DiffFlags.ChangeSetOpts.Mode = ctldiff.ServerSideApplyChangeSetMode
+			}
```

Comment:
I plan to add flag validation step and return error, rather than silently overwriting values provided by user, so it should be a RunE I believe because it can return error.

Reply:
I believe we could move these validations to the `Run()`function  as it returns an error and let the command initialisation have
```go
RunE:    func(_ *cobra.Command, _ []string) error { return o.Run() }
```

Reply:
Moved to PreRunE